src/pendingapproval.js

```diff
@@ -255,6 +255,12 @@ PendingApproval.prototype.approve = function(params, callback) {
   params = params || {};
   common.validateParams(params, [], ['walletPassphrase', 'otp'], callback);
 
+  // We expect retryCount to be undefined when this function is called from the outside
```

Comment:
Since we only intend to retry one time and one time only ..... Instead of recursively calling this fcn and adding logic around persisting retry state .... 

Can we perhaps get away with:

1. Removing this conditional.
2. Removing the call to `self.approve()`

and simply call the recreate tx fcn + return that value in the exception handler like:

line: 325

```javascript

.catch(function(error) {
  // If the user cannot recreate the transaction, raise
  if (!canRecreateTransaction && .... ){
    throw new Error('unspents expired, wallet passphrase or xprv required to recreate transaction');
  }

  // However, if the user can recreate the tx, do so inline and return that value
  // ie: handle recreation + return logic inline; not be calling `self.approve()`
  else if (error.message.indexOf('could not find unspent output for input') !== -1 ) {

    const recreationParams = _.extend({}, params, { txHex: self.info().transactionRequest.transaction }, self.info().transactionRequest.buildParams);

    // delete the previous unspents, because we want to recreate a tx with new ones
    delete recreationParams.unspents; 

    // Recreate + resign the tx and return it
    return self.recreateAndSignTransaction(recreationParams);
  }
  
  // Raise for any other case
  else{
    throw();
  }
}
```